confgenerator/config.go

```diff
@@ -323,117 +555,6 @@ func sliceContains(slice []string, value string) bool {
-// For each field, it collects the yaml name, the value, and whether the value is unset/zero.
-// Fields with the "omitempty" annotation are marked as optional.
-func collectYamlFields(s interface{}) []yamlField {
-	var recurse func(reflect.Value) []yamlField
-	recurse = func(sm reflect.Value) []yamlField {
-		var parameters []yamlField
-		tm := sm.Type()
-		if tm.NumField() != sm.NumField() {
-			panic(fmt.Sprintf("expected the number of fields in %v and %v to match", sm, tm))
-		}
-		for i := 0; i < tm.NumField(); i++ {
-			f := tm.Field(i)
-			t, _ := f.Tag.Lookup("yaml")
-			split := strings.Split(t, ",")
-			n := split[0]
-			annotations := split[1:]
-			if n == "-" {
-				continue
-			} else if n == "" {
-				n = strings.ToLower(f.Name)
-			}
-			v := sm.Field(i)
-			if sliceContains(annotations, "inline") {
-				// Expand inline structs.
-				parameters = append(parameters, recurse(v)...)
-			} else if f.PkgPath == "" { // skip private non-struct fields
-				t, _ := f.Tag.Lookup("validate")
-				validation := strings.Split(t, ",")
-				parameters = append(parameters, yamlField{
-					Name:     n,
-					Required: sliceContains(validation, "required"),
-					Value:    v.Interface(),
-					IsZero:   v.IsZero(),
-				})
-			}
-		}
-		return parameters
-	}
-	return recurse(reflect.ValueOf(s))
-}
-
-func validateParameters(s interface{}, subagent string, kind string, id string, componentType string) error {
-	supportedParameters := supportedParameters[componentType]
-	// Include type when checking.
-	allParameters := []string{"type"}
-	allParameters = append(allParameters, supportedParameters...)
-	additionalValidation, hasAdditionalValidation := additionalParameterValidation[componentType]
-	parameters := collectYamlFields(s)
-	for _, p := range parameters {
-		if !sliceContains(allParameters, p.Name) {
-			if !p.IsZero {
-				// e.g. parameter "transport_protocol" in "files" type logging receiver "receiver_1" is not supported.
-				// Supported parameters: [include_paths, exclude_paths].
-				return fmt.Errorf(`%s is not supported. Supported parameters: [%s].`,
-					parameterErrorPrefix(subagent, kind, id, componentType, p.Name), strings.Join(supportedParameters, ", "))
-			}
-			continue
-		}
-		if p.IsZero && p.Required {
-			// e.g. parameter "include_paths" in "files" type logging receiver "receiver_1" is required.
-			return fmt.Errorf(`%s is required.`, parameterErrorPrefix(subagent, kind, id, componentType, p.Name))
-		}
-		if hasAdditionalValidation {
-			if f, ok := additionalValidation[p.Name]; ok {
-				if err := f(p.Value); err != nil {
-					// e.g. parameter "collection_interval" in "hostmetrics" type metrics receiver "receiver_1"
-					// has invalid value "1s": below the minimum threshold of "10s".
-					return fmt.Errorf(`%s has invalid value %q: %s`, parameterErrorPrefix(subagent, kind, id, componentType, p.Name), p.Value, err)
-				}
-			}
-		}
-	}
-	return nil
-}
-
 var (
 	defaultProcessors = []string{
```

Comment:
Are these 3 vars still necessary?

Reply:
Two of them are still used. Already removed the third.